Jenkinsfile.release

```diff
@@ -0,0 +1,79 @@
+#!groovy
+
+node {
+    properties([
+        parameters([
+            string(
+                defaultValue: '',
```

Comment:
We could also go the route of using `ValidatingStringParameterDefinition`. 

I could see this preventing a scenario where someone leaves this field blank, presses build, doesn't cancel the build, and then the infrastructure is compromised. At the same time, you can still enter the wrong or a non-existent SHA, and do just as much damage.

Another possibility is we require a release tag to be threaded through here, instead of a SHA.

I also feel fine leaving things the way they are.

Reply:
I think it makes sense to add support `ValidatingStringParameterDefinition` such that a valid value needs to be a 7 character alphanumeric value. I tested this regular expression in isolation to achieve that: `^[a-z0-9]{7}$`.

Would it also be valuable to enable the `trim` flag?

Reply:
Sounds good.

`$` matches the end of line, so I don't think this would be necessary.